atomic_reactor/inner.py

```diff
@@ -414,6 +416,16 @@ def build_docker_image(self):
                 logger.error("one or more prepublish plugins failed: %s", ex)
                 raise
 
+            if self.build_result.is_image_available():
+                history = self.builder.tasker.d.history(self.builder.image_id)
+                diff_ids = self.built_image_inspect[INSPECT_ROOTFS][INSPECT_ROOTFS_LAYERS]
+
+                # diff_ids is ordered oldest first
+                # history is ordered newest first
+                # We want layer_sizes to be ordered oldest first
+                self.layer_sizes = [{"diff_id": diff_id, "size": layer['Size']}
+                                    for (diff_id, layer) in zip(diff_ids, reversed(history))]
```

Comment:
By the way, is there a particular reason to do this in `inner.py` instead of `post_koji_upload.py`?

Reply:
apart from the information gathered after squash and be available to other plugins when needed, no

Reply:
Alright. I'd rather have this method (`build_docker_image`) only be concerned with dispatching the plugins for each phase in order, and move the code inspecting layer sizes into koji_upload.

Reply:
Now that I've remembered that exit_koji_promote.py will also need this, maybe it's fine to leave this where it is.

Reply:
koji_promote plugin isn't used since arrangement 3